README.md

```diff
@@ -1,2 +1,101 @@
-# homming-pidgeon
-Write messages reveived from input interface to output interface
+# Homing pigeon
+
+Deliver messages from an input interface to an output interface.
+
+![](images/logo.png)
+
+[Credits](#acknowledgments)
+
+### Overview
+
+![](images/diagram.png)
+
+Homing Pigeon is an application thought to connect read and write adapters easily.
+Whatever comes from an input, it will be sent to a specific output, without the need of writing the usual boilerplate code.
+This tool is thought so that we can easily plug in any kind of adapter for both input and ouput.
+We can use this application to easily store messages, or to easily proxy messages.
+A few examples could be: AMQP to Elasticsearch, HTTP to MySQL, HTTP to HTTP, or any kind of combination.
+
+An important detail, is that the message will be forwarded from the read interface to the write interface "as is", therefore the expected format would depend on which write adapter is connected.
+
+### Currently implemented interfaces
+
+#### Read interfaces
+
+##### RabbitMQ
+Reader interface can handle acks and nacks. Nacks will be automatically sent to dead letter channel withouth retrying
```

Comment:
Read interface reads messages from a single queue and nacks or acks the messages depending on the writer result. All the nacks will be automatically sent to adead letter exchange without retrying